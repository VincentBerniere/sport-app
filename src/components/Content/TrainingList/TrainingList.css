#boards {
    column-width: 250px;
    column-gap: 15px;
    width: 95%;
    max-width: 100%;
}

div#boards .boards-element {
    background: #fefefe;
    margin: 0 2px 15px;
    padding-bottom: 10px;
    transition: opacity .4s ease-in-out;
    display: inline-block;
}

div#boards .boards-element img {
    width: 100%;
    height: auto;
    border-bottom: 1px solid #ccc;
    padding-bottom: 15px;
    margin-bottom: 5px;
}

div#boards .boards-element figcaption {
    font-size: .9rem;
    color: #444;
    line-height: 1.5;
}

div#boards small {
    font-size: 1rem;
    float: right;
    text-transform: uppercase;
    color: #aaa;
}

div#boards small a {
    color: #666;
    text-decoration: none;
    transition: .4s color;
}

div#boards:hover .boards-element:not(:hover) {
    opacity: 0.4;
}

@media screen and (max-width: 750px) {
    #boards {
        column-gap: 0px;
    }

    #boards .boards-element {
        width: 100%;
    }
}
