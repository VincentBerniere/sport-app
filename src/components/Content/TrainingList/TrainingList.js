import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { selectTraining, fetchTrainings } from '../../../actions';
import TrainingCard from './TrainingCard/TrainingCard';
import TrainingDetail from '../TrainingDetail/TrainingDetail';

class TrainingList extends Component {
    componentDidMount() {
        this.props.fetchTrainings();
    }

    renderList() {
        return this.props.trainings.map(training => {
            return (
                <div key={training.id} className="sixteen wide mobile six wide tablet four wide computer column">
                    <div className="training-card" onClick={() => this.props.selectTraining(training)}>
                        <TrainingCard
                            training={training}
                        />
                    </div>
                </div>
            );
        });
    }

    render() {
        return (
            <div className="ui centered grid">
                <div className="row">
                    <Link to="/trainings/new">
                        <button className="ui primary button">
                            Créer un entraînement
                        </button>
                    </Link>
                </div>

                {this.renderList()}

                <TrainingDetail />
            </div>
        );
    }
}

const mapStateToProps = state => {
    return { trainings: state.trainings };
};

export default connect(mapStateToProps, { selectTraining, fetchTrainings })(TrainingList);