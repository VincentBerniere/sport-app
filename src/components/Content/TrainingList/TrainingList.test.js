import React from 'react';
import { shallow } from 'enzyme';
import TrainingList from './TrainingList';

describe('TrainingList', () => {
    let trainingList = shallow(<TrainingList />);
    trainingList.setState({ trainings: [{ id: 0 }, { id: 1 }] });

    it('renders correctly', () => {
        expect(trainingList).toMatchSnapshot();
    });

    it('create card foreach training in list', () => {
        expect(trainingList.find('.training-card').length).toEqual(2);
    });
});