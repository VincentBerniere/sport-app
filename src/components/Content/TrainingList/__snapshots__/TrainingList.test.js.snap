// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainingList renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TrainingList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="row"
        >
          <Link
            to="/trainings/new"
          >
            <button
              className="ui primary button"
            >
              Créer un entraînement
            </button>
          </Link>
        </div>,
        <div
          id="boards"
        >
          <div
            className="boards-element"
          >
            <TrainingCard
              training={
                Object {
                  "id": 0,
                }
              }
            />
          </div>
          <div
            className="boards-element"
          >
            <TrainingCard
              training={
                Object {
                  "id": 1,
                }
              }
            />
          </div>
        </div>,
      ],
      "className": "ui centered grid",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            to="/trainings/new"
          >
            <button
              className="ui primary button"
            >
              Créer un entraînement
            </button>
          </Link>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="ui primary button"
            >
              Créer un entraînement
            </button>,
            "to": "/trainings/new",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Créer un entraînement",
              "className": "ui primary button",
            },
            "ref": null,
            "rendered": "Créer un entraînement",
            "type": "button",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="boards-element"
            >
              <TrainingCard
                training={
                  Object {
                    "id": 0,
                  }
                }
              />
            </div>,
            <div
              className="boards-element"
            >
              <TrainingCard
                training={
                  Object {
                    "id": 1,
                  }
                }
              />
            </div>,
          ],
          "id": "boards",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": <TrainingCard
                training={
                  Object {
                    "id": 0,
                  }
                }
              />,
              "className": "boards-element",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "training": Object {
                  "id": 0,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <TrainingCard
                training={
                  Object {
                    "id": 1,
                  }
                }
              />,
              "className": "boards-element",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "training": Object {
                  "id": 1,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row"
          >
            <Link
              to="/trainings/new"
            >
              <button
                className="ui primary button"
              >
                Créer un entraînement
              </button>
            </Link>
          </div>,
          <div
            id="boards"
          >
            <div
              className="boards-element"
            >
              <TrainingCard
                training={
                  Object {
                    "id": 0,
                  }
                }
              />
            </div>
            <div
              className="boards-element"
            >
              <TrainingCard
                training={
                  Object {
                    "id": 1,
                  }
                }
              />
            </div>
          </div>,
        ],
        "className": "ui centered grid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              to="/trainings/new"
            >
              <button
                className="ui primary button"
              >
                Créer un entraînement
              </button>
            </Link>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="ui primary button"
              >
                Créer un entraînement
              </button>,
              "to": "/trainings/new",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Créer un entraînement",
                "className": "ui primary button",
              },
              "ref": null,
              "rendered": "Créer un entraînement",
              "type": "button",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="boards-element"
              >
                <TrainingCard
                  training={
                    Object {
                      "id": 0,
                    }
                  }
                />
              </div>,
              <div
                className="boards-element"
              >
                <TrainingCard
                  training={
                    Object {
                      "id": 1,
                    }
                  }
                />
              </div>,
            ],
            "id": "boards",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": <TrainingCard
                  training={
                    Object {
                      "id": 0,
                    }
                  }
                />,
                "className": "boards-element",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "training": Object {
                    "id": 0,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": <TrainingCard
                  training={
                    Object {
                      "id": 1,
                    }
                  }
                />,
                "className": "boards-element",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "training": Object {
                    "id": 1,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
